pkg/ccloudv2/flink_gateway.go

```diff
@@ -27,7 +30,22 @@ type FlinkGatewayClient struct {
 func NewFlinkGatewayClient(url, userAgent string, unsafeTrace bool, authToken string) *FlinkGatewayClient {
 	cfg := flinkgatewayv1alpha1.NewConfiguration()
 	cfg.Debug = unsafeTrace
-	cfg.HTTPClient = NewRetryableHttpClient(unsafeTrace)
+	cfg.HTTPClient = NewRetryableHttpClientWithRedirect(unsafeTrace,
+		func(req *http.Request, via []*http.Request) error {
+			// Default net/http implementation allows 10 redirects - https://go.dev/src/net/http/client.go
+			if len(via) >= 10 {
```

Comment:
Should we be a bit more strict here and have a lower threshold like say `5` to give up and declare it as a redirect loop? (I don't foresee a scenario where we need to support more than one redirect in Flink GW)

Reply:
Lowered to 5 to fail fast in case of redirect loops.